homeassistant/components/conversation/trigger.py

```diff
@@ -32,6 +41,8 @@ def has_no_punctuation(value: list[str]) -> list[str]:
         vol.Required(CONF_COMMAND): vol.All(
             cv.ensure_list, [cv.string], has_no_punctuation
         ),
+        vol.Optional(CONF_RESPONSE_SUCCESS): cv.string,
+        vol.Optional(CONF_RESPONSE_ERROR): cv.string,
```

Comment:
This was left out from the original, as we didn't think this was flexible enough.

We want to be able to set a response during automation (and thus allow a single automation to have multiple possible responses).

This PR contains an implementation we, therefore skipped/omitted on purpose.

../Frenck

Reply:
You can set whatever variable you want during an automation to contain whatever value you want and use it in the response, even as the entire response if you want to. Isn't that flexible enough?

Reply:
I saw, it is done by using a template, which is not clear and not easy to use.

What we are looking for, is the possibility of using e.g., a dedicated action.

Reply:
Sorry, but that doesn't make sense to me. You can start an automation from multiple triggers (some sentences, some not) and to have an action that does nothing if no sentence has triggered an automation (or if no sentence trigger even exists) seems... weird. :)

More over, with my approach, each sentence trigger could get a different response, which in some languages makes perfect sense due to noun genders, numbers etc.

Either way, i don't understand what the expectation is, i.e. how i am supposed to alter this PR to have it approved.

Reply:
>  You can start an automation from multiple triggers (some sentences, some not) and to have an action that does nothing if no sentence has triggered an automation (or if no sentence trigger even exists) seems... weird. :)

If there is need for such case, we have trigger IDs, which solve that exact problem.

> More over, with my approach, each sentence trigger could get a different response, which in some languages makes perfect sense due to noun genders, numbers etc.

As said, this approach has been considered and we have decided not to include it.

> i don't understand what the expectation is, i.e. how i am supposed to alter this PR to have it approved.

I don't know either, as the answer to this hasn't been formed (to my knowledge).

Reply:
> As said, this approach has been considered and we have decided not to include it.

OK, but i was sort of hoping for an explanation as to why not, so i wouldn't come up with an alternative solution that's even worse. The flexibility argument, IMHO, does not stand. If anything, this seems like the **more** flexible approach to me.

Reply:
> OK, but i was sort of hoping for an explanation as to why not,

I gave that above?

> The flexibility argument, IMHO, does not stand.

It does, as explained above. Your solution requires the use of templates to give multiple responses, which is unneeded (the useability/easiness part of the flexibility statement in this case).

> If anything, this seems like the more flexible approach to me.

I strongly disagree. In your case, it is set on a trigger. This is the only place you can define it (unless one does magic with templates and setting variables). We avoided this approach for that reason. Instead, we want to be able to set a response anywhere in an automation (allow to set ones, in for example, choose statements).

Reply:
@frenck I have a usecase when per-trigger responses are an option for a custom response and a `Set response` action is not.

It's true that most triggers are used to start an automation, but what about when you `Wait for trigger`? For example, you could wait for user (vocal) feedback in an automation.

Imagine this scenario: you wake up and say "Good morning" to start some morning routine (sentence trigger) and a voice satellite in the bedroom responds with "Hey, I'll open the shutters, turn on the lights, bla bla" (trigger response), then asks you (action 1) if you'd like HA to make you a coffee. It then starts the `voice_assistant` component in ESPHome to listen for a response (action 2). A `Wait for trigger` (`sentence`) action is started (action 3), with possible responses Yes/No. When you respond, based on your response, it either includes making a coffee in your morning routine or not and responds accordingly.

How would a "set response" action work in that case?

Reply:
The same, both this one and the suggested one, need a the script to finish before a response is processed.